libs/community/langchain_community/tools/azure_ai_services/azure_translator/azureTranslate.py

```diff
@@ -0,0 +1,150 @@
+        """
+        Perform text translation using the Azure Translator API.
+
+        Args:
+            text (str): The text to be translated.
+            to_language (str): The target language to translate to.
+
+        Returns:
+            str: The translation result.
+        """
+        if not text:
+            logger.error("Input text for translation is empty.")
+            return None
+
+        body = [{"Text": text}]
+        try:
+            response = self.translate_client.translate(
+                body=body, to_language=[to_language]
+            )
+            if response:
+                logger.warning(
+                    f"Translation successful: {response[0].translations[0].text}"
+                )  # Use WARNING level for successful operations
+                return response[0].translations[0].text
+            else:
+                logger.error("Translation failed with an empty response")
+                return None
+        except Exception as e:
+            logger.error(f"Translation failed: {e}")
+            raise
+
+    def _run(
+        self,
+        query: str,
+        to_language: str,
+        run_manager: Optional[CallbackManagerForToolRun] = None,
+    ) -> str:
+        """
+        Run the tool to perform translation.
+
+        Args:
+            query (str): The text to be translated.
+            to_language (str): The target language to translate to.
+            run_manager (Optional[CallbackManagerForToolRun]): A callback manager
+            for tracking the tool run.
+
+        Returns:
+            str: The translated text.
+        """
+        try:
+            return self._translate_text(query, to_language)
+        except Exception as e:
+            raise RuntimeError(f"Error while running AzureTranslateTool: {e}")
+
+    @classmethod
+    def from_env(cls):
+        """
+        Create an instance of the tool using environment variables.
+        """
+        translate_key = os.getenv("AZURE_OPENAI_TRANSLATE_API_KEY")
+        translate_endpoint = os.getenv("AZURE_OPENAI_TRANSLATE_ENDPOINT")
+
+        if not translate_key:
+            raise ValueError(
+                "AZURE_TRANSLATE_API_KEY is missing in environment variables"
+            )
+
+        if not translate_endpoint:
+            raise ValueError(
+                "AZURE_TRANSLATE_ENDPOINT is missing in environment variables"
+            )
+
+        logger.info(f"API Key: {translate_key[:4]}**** (masked)")
+        logger.info(f"Endpoint: {translate_endpoint}")
+
+        return cls(translate_key=translate_key, translate_endpoint=translate_endpoint)
+
+
+# Example test usage for the AzureTranslateTool
+if __name__ == "__main__":
+    tool = AzureTranslateTool.from_env()
+    try:
+        translated_text = tool._run("good morning, How are you?", "es")
+        logger.info(f"Translated text: {translated_text}")
+    except RuntimeError as e:
+        logger.error(f"Error occurred: {e}")
```

Comment:
It's fine to have this for testing for now, but we'll need to remove it in the final PR